server/routes/companies.js

```diff
@@ -116,25 +179,399 @@ router.get('/filter', function(req, res, next) {
+                 company_id: company_id,
+                 major_id: major_ids[n]
+              })
+            }
+          } else {
+            companyMajors = {
+                 company_id: company_id,
+                 major_id: major_ids
+              }
+          }
+          var queryMajor = knex('company_major').insert(companyMajors)
+          await queryMajor.transacting(trx);
+        }
+
+        if (contact_ids) {
+          let companyContacts = [];
+          if(Array.isArray(contact_ids)){
+            let length = contact_ids.length;
+            for(let n=0; n < length; n++)
+            {
+              companyContacts.push({
+                 company_id: company_id,
+                 contact_id: contact_ids[n]
+              })
+            }
+          } else {
+              companyContacts = {
+                 company_id: company_id,
+                 contact_id: contact_ids
+              }
+          }
+          var queryContact = knex('company_contact').insert(companyContacts)
+          await queryContact.transacting(trx);
+        }   
+
+        if (event_ids) {
+          let companyEvents = [];
+          if(Array.isArray(event_ids)){
+            let length = event_ids.length;
+            for(let n=0; n < length; n++)
+            {
+              companyEvents.push({
+                 company_id: company_id,
+                 event_id: event_ids[n]
+              })
+            }
+          } else{
+              companyEvents = {
+                 company_id: company_id,
+                 event_id: event_ids
+              }
+          }
+          var queryEvent = knex('event_company').insert(companyEvents)
+          await queryEvent.transacting(trx);
+        }
+
+        return trx.commit;
+      })
+    })
+    .then(function(){
+      res.send(util.message('Successfully inserted company'));
+    })
+    .catch(err => {return next(err) });
+  });
+
+// Update a single company
+router.put('/:company_id', function(req, res, next) {
+  values = {
+    name: req.body.name,
+    website: req.body.website,
+    logo: req.body.logo,
+    citizenship_requirement: req.body.citizenship_requirement,
+    description: req.body.description
+  }
+  let ranks = req.body.rank;
+  let user_ids = req.body.user_id;
+  let position_ids = req.body.position_id;
+  let major_ids = req.body.major_id;
+  let contact_ids = req.body.contact_id;
+  let event_ids = req.body.event_id;
+  //let remove_ranks = req.body.remove_rank;
+  let remove_user_ids = req.body.remove_user_id;
+  let remove_position_ids = req.body.remove_position_id;
+  let remove_major_ids = req.body.remove_major_id;
+  let remove_contact_ids = req.body.remove_contact_id;
+  let remove_event_ids = req.body.remove_event_id;
+  let company_id = req.params.company_id;
+
+  console.log(req.body.position_id);
+
+  var queryCompany = knex('company').update(values).where({ id: req.params.company_id })
+  knex.transaction(async function(trx) {
```

Comment:
everything said for post func above applies here too. I'll take another look once those changes are added

Reply:
Implemented 2 for loops for all the post requests. Will work on making it look less cluttered.

Reply:
done!